### PR TITLE
Form.fields_template

### PR DESCRIPTION
**Pull request by berycz**
enhancement by #408

### CODE REVIEW

**Comment by berycz** [comment]
I think this one needs your closer attention, I'm not really sure about some parts
1) I took an example from django and I'm using `hidden_fields` where I also put custom `Field.hidden`'s - I'm not sure whether that is good solution or bad
2) also should there be the `part.input.__html__()` or `part.__html__()`? but the latter one has the wrappers, which may or may not be wanted imo
3) the "trailing" slash in `format_html('<input type="hidden" name="{}" value="{}" />', k, v)` might be removed imo, since `field.input.__html__` does not add it either
4) or maybe instead of hardcoding the hiddens they might be somehow as `Field.hidden`, so then it would fix 1 and 2
5) can you imagine a need for passing groups as instances instead of `.__html__()`? but then how to do it?
6) if you accept this PR, then it might be good to put it in the docs imo

File: iommi/form__tests.py [hunk 1]
```diff
@@ -3579,3 +3579,49 @@ def test_bootstrap_crash_when_no_iommi_name_on_field_group(settings):
...
         iommi_style='bootstrap',
     ).bind(request=req('get')).render_to_response()
+
+
+def test_render_fields_template():
+    class MyForm(Form):
+        class Meta:
+            fields_template = Template('''
+            <div class="row">
+                <div class="col">
```
Comment (boxed) [inline iommi/form__tests.py:3589]: We can loop over fields instead of hardcoding the two names. That's a nicer example.

Comment (berycz) [reply iommi/form__tests.py:3589]: in my bs5 project I use it for 1-2 fields per row, depends on whether it's input|select or textarea and I also use bs cards (with headers) for "fieldsets", so hardcoding seems to me more like a RL example than a loop, but it's up to you

**Comment by boxed** [comment]
I think we should not respect the grouping feature in the template. It seems more reasonable to me to just assert that you don't have grouping. If you are doing manual layout then you are probably not using grouping.

I also suggest that iommi renders the hidden elements at the end outside the template. You always want them anyway, and you don't care about the layout for those, so we might as well take care of that.

**Comment by berycz** [comment]
and what do you think about the custom hidden fields (`Field.hidden`), should they be also outside template as `part.input.__html__()`, or just in `context["fields"]`?

**Comment by boxed** [comment]
Just `input.__html__()` seems reasonable to me.

**Comment by berycz** [comment]
I wonder if someone might actually need to put a hidden input in a specific place in html... like for some JS stuff, e.g. selectors (getting values from inputs placed only in a specific wrapper for some ajax requests) or maybe even changing type from `hidden` to `text` or whatever

I'm starting to not like the idea that you can add any hidden fields in your `Form`, but then you're not free to move them wherever you want in your template, and you **can't** actually put them in the template at all 🤔

Maybe in the end it would be better to say "If you define fields in the Form, no matter what type, you have to put it in the `fields_template` when you use it!" (so leave them in `context["fields"]`) and then automatically add only those hidden inputs that are automatically generated from `request.GET`

File: iommi/form__tests.py (hunk 1 above)
Comment (berycz) [reply iommi/form__tests.py:3590]: I changed it to a more RL example, or do you still want a loop?

**Comment by boxed** [comment]
Yea.. maybe. If the documentation for fields_template has a nice example that does the right thing that people can copy paste to work off of then it's fine. Something like:

```html
{% for field in form.fields %}
    {% if field.hidden %}
        {{ field }}
    {% else %}
         <<stuff here>>
    {% endif %}
{% endfor %}
```

**Comment by berycz** [comment]
eh, how do I pass an album? :)
also feel free to modify the docs

**Pull request merged**